rfcs/0001-notary-integration.md

```diff
@@ -0,0 +1,69 @@
+# Problem
+kpack does not provide users a way to validate the chain of custody for image tags that it builds. 
+One very common way of accomplishing this is through the [Notary](https://github.com/theupdateframework/notary) project.
+
+# Outcome
+kpack integrates with Notary to sign images that it builds so that users are assured of the chain of custody.
+
+# Actions to take
+
+## Enable Image Singing
+To use Notary image signing:
+1. Add a new section to the image config that allows users to configure notary settings:
+  ```
+  notary:
+    v1:
+        host:      <notary server url>
+        secretRef: <secret-name>
+  ```
+2. Add a new container after export that uses the notary go client to sign the image. The presence or absence of the notary section in the image config will determine if the notary signing container should be executed.
+3.
+    * The singing container would use `report.toml` to access the image digest. The image size would need to be added to `report.toml` or computed by the signer container.
+      The `report.toml` must exist in its own volume as export does quite a bit of mangling to the layers volume.
+    * Alternatively, the image digest and size could be computed by pulling the image in the signer container.
+
+## Key Generation and Storage
+kpack will not generate the targets public or private key pair used to sign images.
```

Comment:
does this still require the root key as well, or were you able to drop it?

Reply:
The root key should not be necessary.